superset/assets/src/dashboard/reducers/getInitialState.js

```diff
@@ -149,6 +157,17 @@ export default function(bootstrapData) {
     },
   };
 
+  // store the color scheme as a layout component so we can undo/redo changes
+  layout[COLOR_SCHEME_ID] = {
```

Comment:
could you not use `layout`? `layout` stores actual layout in a flattened tree structure. we can traverse the `layout` attribute to build the whole tree. color_scheme should not ride on this specific data structure.

Reply:
Sure! But if I move it out of layout, changing color schemes will not be an undoable/redoable action based on how the things are set up at the moment. If people don't care about it being undoable/redoable, I can definitely move it out.

Reply:
oh i see, in dashboard metadata it is separated attribute. i am ok with this change. `layout` this name confused me. 👍

Reply:
👍

Reply:
Actually I think I will move it to dashboardState. I explained in reply to your other comment what happens when I do that. What do you think?

Reply:
if move to dashboardState, remember to add logic to update `hasUnsavedChanges` if user change this state. 

I am ok with both solution, undo-able or not undo-able. we used `dashboardLayout` to represent current editing state. if you feel this name is too close to `layout`, you can re-name to `dashboardEditingState` or `dashboardEditingHistory` or anything, and it can contain all redo-able state besides layout changes.

Reply:
I ended up moving it to dashboardState. I would have liked to leave it in the layout object so it stays undoable but it started to leave to a refactoring effort to rename a bunch of things to provide clarity and reduce confusion. I went with the route that required less unrelated changes and maybe in the future this can be refactored in a separate PR.

Reply:
NP. Also i should do this rename work.